src/js/components/List/List.js

```diff
@@ -189,8 +178,8 @@ const List = React.forwardRef(
                     // extract from React's synthetic event pool
                     event.persist();
                     const adjustedEvent = event;
-                    adjustedEvent.item = item;
-                    adjustedEvent.index = index;
+                    adjustedEvent.item = data[active];
```

Comment:
Noticing that this was modified to match what have been in `Keyboard onEnter`, and just confirming that this had no effect on mouse clicks?

Reply:
I wouldn't have thought these changes were needed, since clicking Enter while this item is active would result in this onClick being called, which already has `item` and `index` in scope.

Reply:
The issue was that without updating them, they would pass the incorrect item value in certain scenarios (when mixing mouse and keyboard interactions). In "happy path" scenarios they would not have been needed. 

However, in the scenario where focus is initiated by the mouse, then up/down keyboard actions, then Enter. The item passed to the `onClickItem()` still references the original mouse click item and index. The following screenshots illustrate this scenario w/out the change. 

**Scenario - onEnter removed, onClick w/out change - onClickItem story**
Config:
```
onClick: event => {
  // extract from React's synthetic event pool
  event.persist();
  const adjustedEvent = event;
  adjustedEvent.item = item;
  // adjustedEvent.item = data[active];
  adjustedEvent.index = index;
  // adjustedEvent.index = active;
  onClickItem(adjustedEvent);
},
```
Step 1 
- Load
![Screen Shot 2020-07-08 at 10 19 13 AM](https://user-images.githubusercontent.com/1756948/86944579-30720f00-c105-11ea-9f88-c86a274b1fbd.png)

Step 2
- Click on 'entry-3'
- Expected entry-3 object
- Result entry-3 object
![Screen Shot 2020-07-08 at 10 19 34 AM](https://user-images.githubusercontent.com/1756948/86945942-eb4edc80-c106-11ea-948e-f37044224952.png)

Step 3
- Arrow down to entry-7
- Press Enter
- Expected entry-7 object
- Result entry-3 object
![Screen Shot 2020-07-08 at 10 19 59 AM](https://user-images.githubusercontent.com/1756948/86946137-30730e80-c107-11ea-9ff7-6fa620aed53a.png)

Fail.

----------------------------------------------------

**Scenario - onEnter removed, onClick w/ change - onClickItem story**
Config:
```
onClick: event => {
  // extract from React's synthetic event pool
  event.persist();
  const adjustedEvent = event;
  // adjustedEvent.item = item;
  adjustedEvent.item = data[active];
  // adjustedEvent.index = index;
  adjustedEvent.index = active;
  onClickItem(adjustedEvent);
},
```
Step 1 
- Load
![Screen Shot 2020-07-08 at 10 23 17 AM](https://user-images.githubusercontent.com/1756948/86946505-b1caa100-c107-11ea-9655-75282c0caba6.png)


Step 2
- Click on 'entry-4'
- Expected entry-4 object
- Result entry-4 object
![Screen Shot 2020-07-08 at 10 23 37 AM](https://user-images.githubusercontent.com/1756948/86946515-b727eb80-c107-11ea-9bbf-a251c2464271.png)


Step 3
- Arrow down to entry-8
- Press Enter
- Expected entry-8 object
- Result entry-8 object
![Screen Shot 2020-07-08 at 10 23 53 AM](https://user-images.githubusercontent.com/1756948/86946533-bd1dcc80-c107-11ea-823a-a5b085aa8de9.png)


Pass.

-------------------------------------------------

Reply:
That said, I just uncovered an issue with this setup...  by removing the onEnter altogether, we lost keyboard only functionality... so I need to revisit.

Reply:
Thanks for the detailed examples here and checking out these different cases. By "lost keyboard only functionality", you mean that if you hadn't clicked on an item first with mouse, then you would be able to arrow through the options?

Reply:
> Thanks for the detailed examples here and checking out these different cases. By "lost keyboard only functionality", you mean that if you hadn't clicked on an item first with mouse, then you would be able to arrow through the options?

Arrow through works fine, but arrowing through then pressing Enter does not. 

There is a test written for 'Enter key' and this passed, so I missed manually testing: Tab (for focus on list) --> arrow up/down --> Enter. However, looking closer at the test's implementation, it is initiated with a click, then later presses Enter. I think we need to revisit how these tests are written and ensure they include keyboard only scenarios.